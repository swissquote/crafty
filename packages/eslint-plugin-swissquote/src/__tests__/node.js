/* global describe, it, expect */

const { prepareESLint, lint } = require("../../test_utils");

const engine = prepareESLint("recommended", "node");

it("Doesn't warn on console.log", async () => {
  const result = await lint(engine, `console.log("Yeah");\n`);

  expect(result.messages).toEqual([], "no messages");
  expect(result.warningCount).toBe(0, "no warnings");
  expect(result.errorCount).toBe(0, "no errors");
});

it("Works with ES6", async () => {
  const result = await lint(
    engine,
    `
const something = [];

something.push("a value");

console.log(something);
`
  );

  expect(result.messages).toEqual([], "no messages");
  expect(result.warningCount).toBe(0, "no warnings");
  expect(result.errorCount).toBe(0, "no errors");
});

it("Works with TypeScript", async () => {
  const result = await lint(
    engine,
    `
import useSWROrig from "swr";

export type UnPromisify<T> = T extends Promise<infer U> ? U : T;

export function obviousString(): string {
  const myString: string = "a string";
  console.log(myString);
  return myString;
}

export function useSWR<T, Error = any>(
  url
): {
  data?: T;
  error?: Error;
  revalidate: () => Promise<boolean>;
  isValidating: boolean;
} {
  // Skip it on the server side.
  if (typeof window === undefined) {
    return {
      async revalidate() {
        return true;
      },
      isValidating: true
    };
  }

  return useSWROrig<T, Error>(
    url,
    async (u): Promise<T> => {
      const result = await fetch(u);

      if (result.status == 200) {
        return await result.json();
      }

      throw new Error(result.statusText);
    },
    {}
  );
}
`,
    "utils.ts"
  );

  expect(result.messages).toMatchSnapshot();
  expect(result.warningCount).toBe(0, "no warnings");
  expect(result.errorCount).toBe(3, "no errors");
});
