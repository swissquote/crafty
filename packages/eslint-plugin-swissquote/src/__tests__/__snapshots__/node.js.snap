// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Works with TypeScript 1`] = `
Array [
  Object {
    "column": 9,
    "endColumn": 38,
    "endLine": 6,
    "fix": Object {
      "range": Array [
        155,
        163,
      ],
      "text": "",
    },
    "line": 6,
    "message": "Type string trivially inferred from a string literal, remove type annotation.",
    "messageId": "noInferrableType",
    "nodeType": "VariableDeclarator",
    "ruleId": "@swissquote/swissquote/@typescript-eslint/no-inferrable-types",
    "severity": 2,
  },
  Object {
    "column": 10,
    "endColumn": 20,
    "endLine": 29,
    "line": 29,
    "message": "React Hook \\"useSWROrig\\" is called conditionally. React Hooks must be called in the exact same order in every component render. Did you accidentally call a React Hook after an early return?",
    "nodeType": "Identifier",
    "ruleId": "@swissquote/swissquote/react-hooks/rules-of-hooks",
    "severity": 2,
  },
  Object {
    "column": 25,
    "endColumn": 27,
    "endLine": 34,
    "line": 34,
    "message": "Expected '===' and instead saw '=='.",
    "messageId": "unexpected",
    "nodeType": "BinaryExpression",
    "ruleId": "eqeqeq",
    "severity": 2,
  },
  Object {
    "column": 16,
    "endColumn": 35,
    "endLine": 35,
    "line": 35,
    "message": "Redundant use of \`await\` on a return value.",
    "messageId": "redundantUseOfAwait",
    "nodeType": "Identifier",
    "ruleId": "no-return-await",
    "severity": 2,
  },
]
`;
